### PR TITLE
Enhance newDataSyncService

### PR DESCRIPTION
**Pull request by XuanYang-cn**
- Add flowgraph.Assemble assembles nodes in flowgraph.go
- remove fgCtx in newDataSyncService
- Add newServiceWithEtcdTickler func, reduce param numbers to 3
- Remove unnecessary params
  - node.Session.serverID
  - config.maxQueueLength, config.maxParallelish

See also: https://github.com/milvus-io/milvus/issues/27207

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/datanode/event_manager.go [hunk 1]
```diff
@@ -291,7 +291,7 @@ func isEndWatchState(state datapb.ChannelWatchState) bool {
 		state != datapb.ChannelWatchState_Uncomplete // legacy state, equal to ToWatch
 }
 
-type tickler struct {
+type etcdTickler struct {
```
Comment (jiaoew1991) [inline internal/datanode/event_manager.go:294]: tickler is better than etcdTickler

Comment (XuanYang-cn) [reply internal/datanode/event_manager.go:294]: Will add a new tickler without etcd operation later, etcdTickler will be deprecated soon, so I changed the name.

Comment (jiaoew1991) [reply internal/datanode/event_manager.go:294]: ok

**Comment by XuanYang-cn** [comment]
rerun ut

**Pull request merged**